homeassistant/components/soma/cover.py

```diff
@@ -0,0 +1,81 @@
+"""
+Support for Soma Covers.
+
+For more details about this platform, please refer to the documentation at
+"""
+
+import logging
+
+from homeassistant.components.cover import CoverDevice, ATTR_POSITION
+from homeassistant.components.soma import DOMAIN, SomaEntity, DEVICES, API
+from homeassistant.exceptions import PlatformNotReady
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Soma cover platform."""
+    def get_covers():
+        """Retrieve covers."""
+        devices = hass.data[DOMAIN][DEVICES]
+
+        return [SomaCover(cover, hass.data[DOMAIN][API]) for cover in
+                devices]
+
+    async_add_entities(await hass.async_add_executor_job(get_covers), True)
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Old way of setting up platform.
+
+    Can only be called when a user accidentally mentions the platform in their
+    config. But even in that case it would have been ignored.
+    """
+    pass
+
+
+class SomaCover(SomaEntity, CoverDevice):
+    """Representation of a Soma cover device."""
+
+    async def async_update(self):
+        """Update the device with the latest data."""
+        await super().async_update()
+
+    def close_cover(self, **kwargs):
+        """Close the cover."""
+        if self.api.close_shade(self.device['mac'])['result'] != 'success':
+            raise PlatformNotReady()
```

Comment:
We can't raise `PlatformNotReady` here. We can only raise that in `async_setup_entry`.

Reply:
I didn't know what to throw or how to notify the system of a failed communication attempt. Since we are connecting to a bridge device that in turn connects to BLE devices that may or may not be in range we get occasional communication failures. Throwing this here at least got me a little notification.

What would be a better solution to this?

Reply:
Just log an error.

Reply:
I added errors to log on this event but ideally I would disable the entity until the automatic update gets a response back and only then re-enable. I have to see how I can do that. Or what would be the standard approach to this?

Reply:
We have the `available` property for this purpose. Return `False` when an entity is offline.